plugins/amazonq/codetransform/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codemodernizer/plan/CodeModernizerPlanEditor.kt

```diff
@@ -3,178 +3,495 @@
+                    object : MouseAdapter() {
+                        override fun mouseClicked(e: MouseEvent) {
+                            val row = renderedAppendixTable.rowAtPoint(e.point)
+                            val col = renderedAppendixTable.columnAtPoint(e.point)
+                            if (col == 0) { // means a file name was clicked
+                                val fileClicked =
+                                    LocalFileSystem.getInstance().findFileByPath(
+                                        "${project.basePath}/${renderedAppendixTable.getValueAt(row, col)}",
+                                    )
+                                if (fileClicked != null) {
+                                    FileEditorManager.getInstance(project).openFile(fileClicked, true)
+                                }
+                            }
+                        }
+                    },
+                )
+
+                val tablePanel =
+                    JPanel().apply {
+                        layout = BorderLayout()
+                        add(JBScrollPane(renderedAppendixTable), BorderLayout.NORTH)
+                        border = tableBorder(renderedAppendixTable.rowCount * renderedAppendixTable.rowHeight)
+                    }
+                add(
+                    tablePanel,
+                    constraints.apply {
+                        weighty = 1.0
+                        fill = GridBagConstraints.BOTH
+                    },
+                )
+            }
+        return panel
     }
 
     private fun transformationStepPanel(step: TransformationStep): JPanel {
-        val nameLabel = JLabel(step.name()).apply {
-            font = font.deriveFont(
-                CodeModernizerUIConstants.FONT_CONSTRAINTS.BOLD,
-                CodeModernizerUIConstants.PLAN_CONSTRAINTS.TRANSFORMATION_STEP_TITLE_FONT_SIZE
-            )
-            border = nameBoarder()
+        val nameLabel =
+            JLabel(message("codemodernizer.migration_plan.body.steps_name", step.name())).apply {
+                font =
+                    font.deriveFont(
+                        CodeModernizerUIConstants.FONT_CONSTRAINTS.PLAIN,
+                        CodeModernizerUIConstants.PLAN_CONSTRAINTS.STEP_NAME_FONT_SIZE,
+                    )
+                border = CodeModernizerUIConstants.NAME_BORDER
+            }
+
+        val scrollPanel = createScrollPanel()
+        val descriptionText =
+            JTextArea(step.description()).apply {
+                font = CodeModernizerUIConstants.DESCRIPTION_FONT
+                isEditable = false
+                wrapStyleWord = true
+                lineWrap = true
+                isOpaque = false
+                border = CodeModernizerUIConstants.DESCRIPTION_BORDER
+            }
+
+        val table = tableMapping[step.id()]
+
+        val parsedTable = if (table != null) mapper.readValue(table, PlanTable::class.java) else null
```

Comment:
```suggestion
        val parsedTable = if (table != null) mapper.readValue<PlanTable>(table) else null
```

Reply:
Does not work because `readValue` requires a second parameter.

Reply:
`import com.fasterxml.jackson.module.kotlin.readValue`